### ISSUE TITLE
Add scratch dir. to roger for use in workbench

### ISSUE THREAD

**Issue by dlebauer** [terraref/computing-pipeline#367]
As a user of workbench I would like to be able to write out files without hogging resources or running into file size limitations in a docker image. 

### Proposed (and implemented) Solution

To meet this need, I propose to make a globally read/writeable 'scratch' directory under terraref/users thus:

```sh
mkdir /projects/arpae/terraref/users/scratch
chmod a+rw /projects/arpae/terraref/users/scratch/
```

The name 'scratch' should make it clear that there is no guarantee of persistence.

I implemented this and now it appears that anyone with access can find and write to that directory, and anything written there can be read by all.

I wrote a `foo.csv` file from an rstudio environment on terraref.ndslabs.org and these are the resulting permissions:

```sh
$ ls -alrth /projects/arpae/terraref/users/scratch/
total 0
drwxrwsr-x 9 dlebauer 4.0K Oct 19 14:47 ..
drwxrwsrwx 2 dlebauer 4.0K Oct 19 14:48 .
-rw-r--r-- 1     1000   13 Oct 19 14:48 test.csv
```

### Completion Criteria

* [ ] @jdmaloney and / or @craig-willis please choose one:
   * confirm that this is a sensible solution and close the issue
   * confirm that this is a sufficient temporary solution. Create a new issue to propose a better solution and close this issue
    * recognize a terrible potential result of this, chastize me, and `rm -rf` 
* [ ] Identify any oversights / limitations to this approach.
     * it is okay that files are globally readable. This is necessary b/c each workbench environment may have its own user and an onerous solution like linking roger and workbench accounts is not necessary.
     * someone could use up all of the allocated space. @jdmaloney what are some standard protocols for managing scratch spaces (eg I think stampede purged every two weeks, but would be nice not to be so draconian if it isn't necessary)?

**Comment by craig-willis** [comment]
There are a few things to tease out here that are complexities of workbench:

1. We already create and mount a global shared filesystem for each user's home directory in Workbench. These volumes are mounted into every running container under /home/user and mapped into known volumes in the Docker image.  On the "old" workbench instance, this is a shared Cinder volume. In the new instance, this is a Gluster volume shared across all nodes in the cluster (standard workbench configuration)
1. /gpfs/largeblockFS/projects/arpae/terraref/users is already a scratch directory that is mounted RW into and available to every container under /data/terraref/users. The problem you've solved is that the scratch directory allows all workbench users to write to the same space.
1. Finally, there's the very real problem of UID in the running containers, which is something we're hoping to resolve through integration with LDAP -- but work hasn't started.

An alternative to what you've proposed would be to configure Workbench to use "/gpfs/largeblockFS/projects/arpae/terraref/users" as the mounted shared filesystem for users.  This would eliminate the local Cinder disk or Gluster instance.  Users would be able to share data between ROGER and Workbench easily.  But this doesn't solve the permissions problem.

**Comment by dlebauer** [comment]
@craig-willis, 

1. re /home/user
  * where is the global shared filesystem for each user's home directory? Can this be /gpfs/largeblockFS/projects/arpae/terraref/users/username?
   * I presume user2 can't access user1's `/home/user1`, correct? Having a consistent scratch directory would make analysis scripts more portable / easier to share between user1 and user2.
   * having an environment variable like `MYHOME=/home/user` might be handy
2. if /gpfs/largeblockFS/projects/arpae/terraref/users is mounted RW, shouldn't I be able to `touch /data/terraref/users/foo`? I get 'Permission denied'

**Comment by craig-willis** [comment]
> where is the global shared filesystem for each user's home directory? Can this be /gpfs/largeblockFS/projects/arpae/terraref/users/username?

For now, this is on the workbench system under /var/glfs/global. Yes, it can be on ROGER and I think this is a good idea (but will require system changes)

> I presume user2 can't access user1's /home/user1, correct? 

Within workbench, users can't see each other's directories. However, 
 /gpfs/largeblockFS/projects/arpae/terraref/users will be visible 1) if we retain the mount (i.e., /data/terraref/users, but can disable this going forward) and 2) if they have a login on ROGER.

> Having a consistent scratch directory would make analysis scripts more portable / easier to share between user1 and user2.

Yes, and we've done this before by adding a "shared" directory in Workbench. I agree that this is a good thing.

> having an environment variable like MYHOME=/home/user might be handy

Would NDSLABS_HOME work? It's already there...

> if /gpfs/largeblockFS/projects/arpae/terraref/users is mounted RW, shouldn't I be able to touch /data/terraref/users/foo? I get 'Permission denied'

Back to the UID/GID problem... because you're in a container that isn't running under the terra project credentials (e.g., GID prj_cg_arpae), you'll get a permission denied error unless you create a directory via ROGER and open permissions (as you did with scratch).  The best way to deal with this is for workbench to run with shared UID/GID or at the very least as a common user (e.g., workbench) with the prj_cg_arpae GID.  I wish I didn't even have to write this in a comment.

**Comment by craig-willis** [comment]
@dlebauer Spoke with JD about this and he sees no problem with us using the ROGER scratch space for Workbench storage. 

I started configuring the system to test this and ran into an immediate problem trying to migrate user data.  

In the near term, I would like to continue using the Workbench internal shared storage with the users directory mounted with the goal of moving to this model after we've resolved the UID/GID problem.  I hope to have a solution implemented in Workbench in the coming months.
